content/1_General/Contributing.mdx

```diff
@@ -195,4 +195,24 @@ Additional Notes:
   print(a + b)
   ```
 
-  It doesn't matter which one you use (`a+b` or `a + b`), but pick one and stick with it!
\ No newline at end of file
+  It doesn't matter which one you use (`a+b` or `a + b`), but pick one and stick with it!
+
+<LanguageSection>
+
+<CPPSection />
+
+<JavaSection>
+
+<Info title="Java I/O">
+
+`Scanner` is [significantly slower](/general/fast-io?lang=java#fast-input) than `BufferedReader` and should be avoided.
+
+To improve readability, Java solutions using `BufferedReader` should use a wrapper class around input (preferably [`Kattio`](/general/io?lang=java#io-template) for consistency, or use your own variant if you really want). 
+
+</Info>
+
+</JavaSection>
+
+<PySection />
```

Comment:
Maybe mention pep-8 for python?

Reply:
you can, I haven't actually looked at it

Reply:
actually, on second thought, let's just not mention it in case there's some rule that doesn't make sense for competitive programming. We don't get many (any) python PR's anyway :P